src/main/scala/io/iohk/ethereum/jsonrpc/client/RpcBaseClient.scala

```diff
@@ -0,0 +1,102 @@
+package io.iohk.ethereum.jsonrpc.client
+
+import java.io.{PrintWriter, StringWriter}
+import java.util.UUID
+
+import akka.actor.ActorSystem
+import akka.http.scaladsl.model._
+import akka.http.scaladsl.unmarshalling.Unmarshal
+import akka.http.scaladsl.{ConnectionContext, Http, HttpsConnectionContext}
+import io.circe.generic.auto._
+import io.circe.parser.parse
+import io.circe.syntax._
+import io.circe.{Decoder, Json}
+import io.iohk.ethereum.jsonrpc.JsonRpcError
+import io.iohk.ethereum.utils.Logger
+import javax.net.ssl.SSLContext
+import monix.eval.Task
+
+import scala.concurrent.duration._
+import scala.concurrent.{Await, ExecutionContext}
+
+abstract class RpcBaseClient(node: Uri, maybeSslContext: Option[SSLContext])(implicit
```

Comment:
(minor) why not just `RpcClient` ?

Reply:
We had the class RpcClient in the “mallet” module. We have pending remove this module in task:https://jira.iohk.io/browse/ETCM-423
So, I changed the origin class with the same name (now is called RpcClientMallet), because it will be removed.